ngraph/frontend/paddlepaddle/src/op/reduce_ops.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <node_context.hpp>
+#include "default_opset.hpp"
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+
+template <typename T>
+NamedOutputs reduce_ops(const NodeContext& node) {
+    auto x = node.get_ng_input("X");
+    auto keep_dim = node.get_attribute<bool>("keep_dim");
+    auto reduce_all = node.get_attribute<bool>("reduce_all", false);
+    Shape input_shape = x.get_shape();
+    size_t input_rank = input_shape.size();
+    std::vector<int32_t> dims;
+    if (reduce_all) {
+        for (int i = 0; i < (int)input_rank; ++i) {
+            dims.push_back(i);
+        }
+    } else {
+        dims = node.get_attribute<std::vector<int32_t>>("dim");
+    }
+    for (auto i : dims) {
```

Comment:
Can you build a negative test case for it? If nGraph checks the violation, we can remove this check here.